### ISSUE TITLE
Improve error message on workflow file changes

### ISSUE THREAD

**Issue by korthout** [korthout/backport-action#368]
GitHub actions are not allowed to push changes to GitHub action workflow files. When you attempt to backport a pull request with this action that includes changes to a workflow file you'll see an error message that the push failed with exit code 1.

This error code is not helpful enough. We can include the reason for the error and the manual instructions to cherry pick the commits by hand.

**Comment by macmiranda** [comment]
Hey @korthout hitting the exact same issue. Would you know what's the actual reason for this?

**Comment by macmiranda** [comment]
Thanks. It's a shame we can't just add the capability to the runner token

**Comment by korthout** [comment]
Absolutely agree. I guess GitHub sees a vulnerability, but it doesn't really apply to backport-action as it only deals with merged pull requests. 

Let me know whether the workaround works for you, and I'd love to hear any other feedback on the action by the way. :bow:

**Comment by macmiranda** [comment]
Yes, the workaround worked fine. I don't like using PATs though, especially the classic ones, they are overly powerful. It also doesn't make sense to have a PAT for a repo under an organization because the person could leave the company at any time and then the PAT is worthless (we use an automation user for that, but it takes a paid seat of the organization 😞)

Anyway, cool GitHub action. Thanks for sharing it. 

Looking forward to a solution for https://github.com/korthout/backport-action/issues/39

It would also be cool if there was an option to merge the backported commit directly, without the PR.

**Comment by T0biii** [comment]
> Yes, the workaround worked fine. I don't like using PATs though, especially the classic ones, they are overly powerful. It also doesn't make sense to have a PAT for a repo under an organization because the person could leave the company at any time and then the PAT is worthless (we use an automation user for that, but it takes a paid seat of the organization 😞)
> 
> Anyway, cool GitHub action. Thanks for sharing it.
> 
> Looking forward to a solution for [#39](https://github.com/korthout/backport-action/issues/39)
> 
> It would also be cool if there was an option to merge the backported commit directly, without the PR.

Since we don't like PATs either, we use a Github App to make this possible

you only need the app ID and a private key from the app that has workflow permissions on the repo
we set the app token as a repository var and the private key as a repository secret
then you can use this action: `actions/create-github-app-token` to get the github token to use with checkout

see here for app creation: https://github.com/freifunkMUC/site-ffm/pull/516#issue-2631235892

**Comment by macmiranda** [comment]
I have since moved to using a helper bot as well
